python/ray/util/collective/types.py

```diff
@@ -61,9 +61,12 @@ class TensorTransportMetadata:
 
     Args:
         tensor_meta: A list of tuples, each containing the shape and dtype of a tensor.
+        tensor_device: The device of the tensor. Currently, we assume all tensors in the
+        list have the same device type.
     """
 
     tensor_meta: List[Tuple["torch.Size", "torch.dtype"]]
+    tensor_device: Optional["torch.device"] = None
```

Comment:
When is `torch_device` not set, and what is the behavior if `tensor_device` is None?

Reply:
tensor_device will only be None if transferring pure cpu data. otherwise, it will not be none.